pkg/component/gardener/dashboard/config/config.go

```diff
@@ -41,3 +46,37 @@ type UnreachableSeeds struct {
 type ContentSecurityPolicy struct {
 	ConnectSources []string `yaml:"connectSrc"`
 }
+
+// Terminal is the configuration for the terminals.
+type Terminal struct {
+	Container                  TerminalContainer                   `yaml:"container"`
+	ContainerImageDescriptions []TerminalContainerImageDescription `yaml:"containerImageDescriptions"`
+	GardenTerminalHost         TerminalGardenHost                  `yaml:"gardenTerminalHost"`
+	Garden                     TerminalGarden                      `yaml:"garden"`
+}
+
+// TerminalContainer is the configuration for a terminal container.
+type TerminalContainer struct {
+	Image string `yaml:"image"`
+}
+
+// TerminalContainerImageDescription is the configuration for terminal image descriptions.
+type TerminalContainerImageDescription struct {
+	Image       string `yaml:"image"`
+	Description string `yaml:"description"`
+}
```

Comment:
https://github.com/gardener/dashboard/blob/ad62da8019e708ca8639c8c26f5cf5b89e20fc9c/charts/gardener-dashboard/values.yaml#L426C1-L432C87
```yaml
  # terminal:
  #   container:
  #     image: europe-docker.pkg.dev/gardener-project/releases/gardener/ops-toolbelt:latest
  #   containerOperator:
  #     image: ~ # if not defined, value of terminal.container.image will be used
  #   containerImageDescriptions:
  #   - image: /europe-docker.pkg.dev/gardener-project/releases/gardener/ops-toolbelt:.*/ # regexp must start and end with '/', otherwise it's an exact match
  #     description: Run `ghelp` to get information about installed tools and packages
  ```
The container image is the default for the terminal pod; however, it can be changed on the UI. Hence, you can define different help texts/descriptions for different images. `containerImageDescriptions[].image` can also be a regexp

Reply:
From my perspective, this looks like a historically grown API. Would it make sense to streamline it?

Reply:
Opened https://github.com/gardener/dashboard/issues/1825.
Let's continue the discussion there.